hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/InternalExecutionService.java

```diff
@@ -18,14 +18,27 @@
 
 import com.hazelcast.spi.ExecutionService;
 
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
 /**
  * The internal {@link ExecutionService}. Currently this method doesn't contains any additional methods, but
  * in the future they can be added here.
  */
 public interface InternalExecutionService extends ExecutionService {
 
+    ExecutorService getDurable(String name, boolean scheduled);
 
     void executeDurable(String name, Runnable command);
 
+    ScheduledFuture<?> scheduleDurable(String name, Runnable command, long delay, TimeUnit unit);
+
+    <V> ScheduledFuture<V> scheduleDurable(String name, Callable<V> command, long delay, TimeUnit unit);
+
+    ScheduledFuture<?> scheduleDurableWithRepetition(String name, Runnable command,
```

Comment:
Seems `scheduleDurableWithRepetition` may cause some problems. For example: imagine  i have a task which is scheduled to run in every 1 second and my thread-pool for scheduler has only 1 thread. 
But my task run takes longer than 1 second say 10 seconds. In this case, even my task is not finished, scheduler will continue to add new tasks to task queue. Eventually in time whole queue will be occupied and there will be no place for new tasks.

An example of queue size changes after 10 second periods will be something like this:
1 (1)
1 + 10 (11)
1 + 9 + 10 (20)
1 + 8 + 10 + 10 (29)
...

Will this be a problem for the end user WDYT?